public/app/plugins/panel/canvas/editor/element/elementEditor.tsx

```diff
@@ -120,21 +123,28 @@ export function getElementEditor(opts: CanvasEditorOptions): NestedPanelOptions<
         optionBuilder.addBorder(builder, ctx);
       }
 
+      const oneClickModeOptions = [
+        { value: OneClickMode.Off, label: capitalize(OneClickMode.Off) },
+        { value: OneClickMode.Link, label: capitalize(OneClickMode.Link) },
+      ];
+
+      if (actionsEnabled) {
+        oneClickModeOptions.push({ value: OneClickMode.Action, label: capitalize(OneClickMode.Action) });
+      }
+
       builder.addRadio({
-        category: ['Data links'],
+        category: actionsEnabled ? ['Data links and actions'] : ['Data links'],
         path: 'oneClickMode',
         name: 'One-click',
-        description: 'When enabled, a single click opens the first link',
+        description: 'When enabled, a single click opens the first link' + actionsEnabled ? ' or action' : '',
         settings: {
-          options: [
-            { value: OneClickMode.Off, label: capitalize(OneClickMode.Off) },
-            { value: OneClickMode.Link, label: capitalize(OneClickMode.Link) },
-          ],
+          options: oneClickModeOptions,
         },
         defaultValue: OneClickMode.Off,
       });
 
       optionBuilder.addDataLinks(builder, ctx);
+      optionBuilder.addActions(builder, ctx);
```

Comment:
MIght be cleaner / easier to understand in the future to just separate these changes into a single if block  - i.e.

```
const oneClickModeOptions = [...]
let oneClickCategory = "Data links"
let description = etc
if (actionsEnabled) {
  push action to oneClickModeOptions
  oneClickCategory += " and actions"
 description += " or action"
}

builder.addRadio({...})
  ```